### PR TITLE
Feature/multiple aes siv ads

### PR DESCRIPTION
**Pull request by ptsiewie**
# Description

The AES SIV algorithm implementation in WolfSSL would accept only exactly one vector of associated data, even though the definition of the algorithm of RFC5297 allows any number no larger than 126.

Fixes zd#18509

# Testing

The tests in test.c have been extended to include the two examples given in the RFC5297 document. The second of these two examples uses two ADs.

# Checklist

 - [x] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

**Comment by dgarske** [comment]
Okay to test. Contributor agreement in progress.

File: wolfcrypt/src/aes.c [hunk 1]
```diff
@@ -13635,32 +13635,49 @@ static WARN_UNUSED_RESULT int S2V(
...
     }
 
     if (ret == 0) {
-        if (nonceSz > 0) {
-            ShiftAndXorRb(tmp[0], tmp[1]);
-            ret = wc_AesCmacGenerate(tmp[1], &macSz, nonce, nonceSz, key,
-                                     keySz);
+        /* Loop over authenticated associated data AD1..ADn */
+        byte tmpi = 0;
+        for(word32 ai = 0; ai < numAssoc; ++ai) {
```
Comment (dgarske) [inline wolfcrypt/src/aes.c:13657]: For portability please declare `word32 ai` on line above. Code consistency please use `for (` add space.

File: wolfcrypt/src/aes.c [hunk 2]
```diff
@@ -13826,7 +13843,8 @@ int wc_AesSivEncrypt(const byte* key, word32 keySz, const byte* assoc,
                      word32 assocSz, const byte* nonce, word32 nonceSz,
                      const byte* in, word32 inSz, byte* siv, byte* out)
 {
-    return AesSivCipher(key, keySz, assoc, assocSz, nonce, nonceSz, in, inSz,
+    const AesSivAssoc ad0 = { .assoc = assoc, .assocSz=assocSz };
```
Comment (dgarske) [inline wolfcrypt/src/aes.c:13846]: This syntax is not very portable. Please use:
```
AesSivAssoc ad0;
ad0.assoc = assoc;
ad0.assocSz = assocSz;
```

**Review by dgarske — CHANGES_REQUESTED** [review]
Over to @wolfssl-bot and @SparkiDev for further review. Thank you for adding a test case.

**Comment by dgarske** [comment]
Okay to test. Contributor agreement in progress. @ptsiewie sorry for the delay in getting your agreement approved. Expect it next week.

**Review by dgarske — DISMISSED** [review]
Feature looks good to me. @SparkiDev please review. Also should we wrap any of this new SIV support in a new build macro?

**Review by SparkiDev — DISMISSED** [review]

**Comment by SparkiDev** [comment]
retest this please

**Comment by SparkiDev** [comment]
The old API goes through the new API so we will always include code.

**Comment by SparkiDev** [comment]
Hi @ptsiewie 

Errors in ctidy check

TLDR: change '1u' to '1U'

Error:
 shellcheck configure.../var/lib/jenkins/workspace/PRB-multi-test-script/wolfssl/wolfcrypt/src/aes.c:13850:43: warning: integer literal has suffix 'u', which is not uppercase [readability-uppercase-literal-suffix]
return AesSivCipher(key, keySz, &ad0, 1u, nonce, nonceSz, in, inSz,
^~
U
/var/lib/jenkins/workspace/PRB-multi-test-script/wolfssl/wolfcrypt/src/aes.c:13864:43: warning: integer literal has suffix 'u', which is not uppercase [readability-uppercase-literal-suffix]
return AesSivCipher(key, keySz, &ad0, 1u, nonce, nonceSz, in, inSz,
^~
U

**Comment by dgarske** [comment]
Okay to test. Retest this please

**Comment by dgarske** [comment]
Reminder, contributor agreement in progress, but not yet approved. Person needed for that is on vacation this week.

**Comment by dgarske** [comment]
Contributor agreement approved and on file. Thank you @ptsiewie!

**Review by SparkiDev — APPROVED** [review]

**Review by dgarske — APPROVED** [review]

**Comment by ptsiewie** [comment]
Do I need to do something before this branch will be merged?
